### DISCUSSION TITLE
How to interpret the non-self intersection precondition for `CGAL::Polygon_mesh_processing::clip()`?

### DISCUSSION THREAD

**Discussion by ton** [CGAL/cgal#6703] [Q&A] [answered]
This is not an issue report necessarily, although it could be framed as a request to improve the documentation around the preconditions on the input mesh preconditions for `CGAL::Polygon_mesh_processing::clip()`.

Right now, the preconditions state that the input meshes should be non-self intersecting. I have been using the `clip()` operation successfully on input meshes that are self-intersecting, so how strict is this precondition? Can it be defined more accurately when the algorithm breaks down in case of self-intersecting inputs?

In our case, self-intersection exists because of duplicated vertices introduced to resolve the non-manifoldness of a mesh. Is it possible for the algorithm to ever break down in case of self-intersection caused only by duplicated vertices or edges?

**Comment by sloriot** [comment, marked as answer]
The only self-intersections that are an issue are degenerate faces and self-intersecting triangles that are intersected by the clipper.
